### ISSUE TITLE
Are we suppose to set the values of INVALID_UV in the uv_map?

### ISSUE THREAD

**Issue by saslamsameja** [JustusThies/NeuralVoicePuppetry#30]
In neural renderer code there's a line:

mask = (self.input_uv[:,0:1,:,:] == INVALID_UV) & (self.input_uv[:,1:2,:,:] == INVALID_UV)
And INVALID_UV is defined as "-1".

Are we supposed to set the empty region or black region values in uv_map to "-1". Or equivalently set the INVALID_UV=0.0?

**Comment by JustusThies** [comment]
Hi,
yes, empty regions should be flagged with -1.
Best,
Justus Thies

**Comment by saslamsameja** [comment]
Hi @JustusThies , thanks for the clarification. I wanted to confirm one thing. The following lines in face_dataset.py are performing normalization on "TARGET" and "UV" before passing to the "DynamicNeuralTextures_model":

#print('GET ITEM: ', index)
img_fname = os.path.join(self.image_dir, str(image_id).zfill(5) + '.png')
img_numpy = np.asarray(Image.open(img_fname))
TARGET = 2.0 * transforms.ToTensor()(img_numpy.astype(np.float32))/255.0 - 1.0

uv_fname = os.path.join(self.uvs_dir, str(uv_id).zfill(5) + '.exr')
uv_numpy = util.load_exr(uv_fname)
UV = transforms.ToTensor()(uv_numpy.astype(np.float32))
UV = torch.where(UV > 1.0, torch.zeros_like(UV), UV)
UV = torch.where(UV < 0.0, torch.zeros_like(UV), UV)
UV = 2.0 * UV - 1.0

In the above lines I can see that TARGET is normalized but how are we updating the "UV". I think after this processing we will only be left with 1s and -1s in the UV map. Is that the intention here? And how much likey it is that a Rasterized image of UV map has 1s in its pixel values. And finally, let's say I have tuples of (UV map, Image) where UV_map is also a .jpg image and I replace the black area in the UV_map with -1 will this translate to the operations performed in the above lines. And I apply the above operations on TARGET(image) to normalize it. Following is an example of my UV_map:

![00000](https://user-images.githubusercontent.com/32391022/105397214-94dc6200-5c42-11eb-834f-9001801a32a0.jpg)


Or will I need a .exr file and need to perform the above steps?

Thanks.

**Comment by saslamsameja** [comment]
Just in case if anyone is curious, the above-mentioned preprocessing worked for me.
